File: frontend/src/types/screening.ts
// Shared TypeScript Interfaces mirroring Backend Pydantic Schemas

export type SeverityLevel = "LOW" | "MEDIUM" | "HIGH";
export type RiskLevel = "CLEAR" | "REVIEW" | "HIGH_RISK";
export type OfficerAction = "PENDING" | "ACCEPT_CLEAR" | "SEND_TO_SECONDARY_REVIEW" | "REJECT_FRAUD";

export interface BoundingBox {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface ExtractedField {
  field_name: string;
  value: string;
  confidence: number;
  bounding_box?: BoundingBox;
}

export interface ExtractionResult {
  document_category: string;
  category_confidence: number;
  full_name?: ExtractedField;
  date_of_birth?: ExtractedField;
  document_number?: ExtractedField;
  nationality?: ExtractedField;
  gender?: ExtractedField;
  expiry_date?: ExtractedField;
  has_portrait: boolean;
  portrait_image_path?: string;
}

export interface ReasonCode {
  code: string;
  description: string;
  severity: SeverityLevel;
  module_source: string;
  weight: number;
}

export interface RiskAssessment {
  risk_score: number; // 0 - 100
  risk_level: RiskLevel;
  reason_codes: ReasonCode[];
  top_reasons: string[];
  authenticity_score: number;
  validity_score: number;
  identity_score: number;
  requires_manual_inspection: boolean;
}

export interface ScreeningResponse {
  screening_id: string;
  status: "PENDING" | "PROCESSING" | "COMPLETED" | "FAILED";
  timestamp_utc: string;
  processing_time_ms: number;
  risk_assessment: RiskAssessment;
  officer_action_state: OfficerAction;
  officer_statement: string;
}

File: backend/app/schemas/screening.py
from enum import Enum
from typing import Optional, List, Dict, Any
from datetime import datetime, timezone
from pydantic import BaseModel, Field
from app.schemas.document import ValidatedInputDocument
from app.schemas.evidence import EvidenceBundle
from app.schemas.hypothesis import HypothesisResult
from app.schemas.risk import RiskAssessment

class ScreeningStatus(str, Enum):
    PENDING = "PENDING"
    PROCESSING = "PROCESSING"
    COMPLETED = "COMPLETED"
    FAILED = "FAILED"

class OfficerAction(str, Enum):
    PENDING = "PENDING"
    ACCEPT_CLEAR = "ACCEPT_CLEAR"
    SEND_TO_SECONDARY_REVIEW = "SEND_TO_SECONDARY_REVIEW"
    REJECT_FRAUD = "REJECT_FRAUD"

class ScreeningResponse(BaseModel):
    screening_id: str
    status: ScreeningStatus
    timestamp_utc: datetime = Field(default_factory=lambda: datetime.now(timezone.utc))
    processing_time_ms: float = 0.0
    
    # Core results
    document_info: Optional[ValidatedInputDocument] = None
    risk_assessment: Optional[RiskAssessment] = None
    hypothesis_result: Optional[HypothesisResult] = None
    evidence_bundle: Optional[EvidenceBundle] = None
    
    officer_action_state: OfficerAction = OfficerAction.PENDING
    officer_statement: str = "AI ASSISTS • OFFICER DECIDES"
